digitalocean/resource_digitalocean_image.go

```diff
@@ -0,0 +1,312 @@
+package digitalocean
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/digitalocean/godo"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceDigitalOceanImage() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDigitalOceanImageCreate,
+		Read:   resourceDigitalOceanImageRead,
+		Update: resourceDigitalOceanImageUpdate,
+		Delete: resourceDigitalOceanImageDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceDigitalOceanImageImport,
+		},
+		SchemaVersion: 1,
+
+		Schema: map[string]*schema.Schema{
+
+			"name": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Description:   "name of the image",
+				ValidateFunc:  validation.NoZeroValues,
+				ConflictsWith: []string{"slug"},
+			},
+			"slug": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "slug of the image",
+				ValidateFunc: validation.NoZeroValues,
+			},
+			// computed attributes
+			"image": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "slug or id of the image",
+			},
+			"distribution": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "distribution of the OS of the image",
+			},
+			"private": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Is the image private or non-private",
+			},
+			"min_disk_size": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "minimum disk size required by the image",
+			},
+			"regions": {
```

Comment:
For a custom image, this should be user defined. When first creating a custom image, you can specify the initial region.

Images can then be copied to additional regions:
https://developers.digitalocean.com/documentation/v2/#transfer-an-image